### PR TITLE
fields can be reserved words

### PR DESCRIPTION
**Pull request by gw000**
allowed: `x = {debug: 1, type: 2, new: 3} and x.new = x.debug + x.type`
not allowed:  `new(x);` `foo(debug);`

PR checklist:

- [x] Added changelog entry.

### CODE REVIEW

File: polar-core/src/polar.lalrpop [hunk 1]
```diff
@@ -98,6 +114,11 @@ Boolean: Value = <b:"Boolean"> => {
 
 Name: Symbol = <s:"Symbol"> => s;
 
+ResName: Symbol = {
```
Comment (gkaemmer) [inline polar-core/src/polar.lalrpop:117]: Is this used anywhere?

Comment (gw000) [reply polar-core/src/polar.lalrpop:117]: 😮 no good catch ty!

File: polar-core/src/polar.lalrpop [hunk 2]
```diff
@@ -126,6 +142,30 @@ Call: Value = {
     }
 };
 
+DotCall: Value = {
```
Comment (gj) [inline polar-core/src/polar.lalrpop:145]: I think we could probably reduce the duplication between `Call` and `DotCall` with a generic param, but I'm not sure it's worth the effort. Up to you whether you want to experiment with that or just ship it.

File: docs/content/any/project/changelogs/NEXT.md [hunk 3]
```diff
@@ -48,5 +48,6 @@ Link to [relevant documentation section]().
 #### Other bugs & improvements
 
 - The debugger can now break on rule matches.
+- Reserved words can be used as field and method names in dictionaries and objects.
```
Comment (gj) [inline docs/content/any/project/changelogs/NEXT.md:51]: Maybe give some examples of what reserved words are — I don't think we use that term anywhere else in the docs. Maybe "Reserved Polar keywords" would be more descriptive?

**Review by gj — APPROVED** [review]

File: docs/content/any/project/changelogs/NEXT.md (hunk 3 above)
Comment (gw000) [reply docs/content/any/project/changelogs/NEXT.md]: 👍  done

File: polar-core/src/polar.lalrpop (hunk 2 above)
Comment (gw000) [reply polar-core/src/polar.lalrpop:145]: Good call, I'll revisit this later, not familiar enough with lalrpop to know how to do it right now.

**Pull request merged**